test/runtime/local/kernels/EigenCalTest.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 The DAPHNE Consortium
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <runtime/local/datagen/GenGivenVals.h>
+#include <runtime/local/datastructures/DataObjectFactory.h>
+#include <runtime/local/datastructures/DenseMatrix.h>
+#include <runtime/local/kernels/EigenCal.h>
+#include <runtime/local/kernels/CheckEq.h>
+
+#include <tags.h>
+
+#include <catch.hpp>
+
+#include <vector>
+
+#include <cstdint>
+
+
+template<class DT>
+void checkEigenCal(const DT * inMat, const DT * exp1, const DT * exp2) {
+    DT * res1 = nullptr;
+    DT * res2 = nullptr;
+    eigenCal<DT,DT, DT>(res1, res2, inMat, nullptr);
+    CHECK(*res1 == *exp1);
+    CHECK(*res2 == *exp2);
```

Comment:
These checks fail due to precision problems. Use something like this (taken from SolveTest.cpp):
    CHECK(checkEqApprox(res, exp, 1e-6, nullptr));